File: Q2/src/main/java/com/yourorganization/Shipment/Customer/Customer.java
package com.yourorganization.Shipment.Customer; // Package for shipment-related classes

public class Customer {
    private String id; // Identifier for the customer
    private String name; // Customer's name
    private int distanceFromWarehouse; // Distance to warehouse in km

    // Constructor for creating a new Customer
    public Customer(String id, String name, int distanceFromWarehouse) {
        this.id = id; // Setting customer's ID
        this.name = name; // Setting customer's name
        this.distanceFromWarehouse = distanceFromWarehouse; // Setting distance from warehouse
    }

    // Getters and setters for Customer properties
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getDistanceFromWarehouse() {
        return distanceFromWarehouse;
    }

    public void setDistanceFromWarehouse(int distanceFromWarehouse) {
        this.distanceFromWarehouse = distanceFromWarehouse;
    }
}

File: Q3_Cpp/src/Shipment/Customer/Customer.cpp
#include "../../../include/Shipment/Customer/Customer.h"

// Constructor for creating a new Customer
Customer::Customer(std::string id, std::string name, int distanceFromWarehouse)
    : id(id), name(name), distanceFromWarehouse(distanceFromWarehouse) {
}

// Getters and setters for Customer properties
std::string Customer::getId() const {
    return id;
}

void Customer::setId(const std::string& id) {
    this->id = id;
}

std::string Customer::getName() const {
    return name;
}

void Customer::setName(const std::string& name) {
    this->name = name;
}

int Customer::getDistanceFromWarehouse() const {
    return distanceFromWarehouse;
}

void Customer::setDistanceFromWarehouse(int distanceFromWarehouse) {
    this->distanceFromWarehouse = distanceFromWarehouse;
}
